SpiceSharp/Components/Waveforms/Pwl.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using SpiceSharp.IntegrationMethods;
+using SpiceSharp.Simulations;
+
+namespace SpiceSharp.Components.Waveforms
+{
+    public class Pwl : Waveform
+    {
+        public Pwl(double[] times, double[] voltages, double initialVoltage = 0.0)
+        {
+            if (times == null)
+            {
+                throw new NullReferenceException(nameof(times));
+            }
+
+            if (voltages == null)
+            {
+                throw new NullReferenceException(nameof(voltages));
+            }
+
+            if (times.Length != voltages.Length)
+            {
+                throw new InvalidOperationException("PWL - times array has different length than voltages array");
+            }
+
+            Points = CreatePoints(times, voltages, initialVoltage);
+            LineParameters = CreateLineParameters(Points);
```

Comment:
I'd prefer not caching the line slopes and intercepts. If you really want to cache, you can do so when a breakpoint occurs. The parameters won't change until the next breakpoint anyway...

Reply:
@svenboulanger I will not cache.